### PR TITLE
Events for state dump to socket

### PR DESCRIPTION
**Pull request by anurag6**
Yet to add L2_LEARN events. Raising PR to get conversation started.

### CODE REVIEW

**Comment by grafnu** [comment]
Should be against events2 branch to avoid confusion with upstream FaucetSDN PR.

File: faucet/faucet_event.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def __init__(self, socket_path, metrics, logger):
         self.thread = None
         self.lock = NonBlockLock()
         self.event_q = eventlet.queue.Queue(120)
+        self.conn_handlers = []
```
Comment (grafnu) [inline faucet/faucet_event.py]: new_conn_handlers?

File: faucet/faucet_event.py [hunk 2]
```diff
@@ -94,6 +96,14 @@ def _loop(self, sock, _addr):
         except (socket.error, IOError):
             pass
 
+    def register_conn_handler(self, handler):
+        """Registers connection handler to be called on new connection"""
+        self.conn_handlers.append(handler)
+
+    def _new_conn_handler(self):
```
Comment (grafnu) [inline faucet/faucet_event.py]: _call_new_conn_handlers?

File: faucet/valve.py [hunk 3]
```diff
@@ -138,6 +139,35 @@ def close_logs(self):
...
+        self._dump_dp_of_status_event_sock()
+        self._dump_lag_status_event_sock()
+
+    def _dump_ports_status_event_sock(self):
+        port_status = {
+            str(port.number): (port.running()) for port in self.dp.ports.values()}
+        self._notify({'PORTS_STATUS': port_status})
+
+    def _dump_dp_of_status_event_sock(self):
+        if self.dp.dyn_running:
```
Comment (grafnu) [inline faucet/valve.py]: simplify code with intermediate variable for reason

File: faucet/valves_manager.py [hunk 4]
```diff
@@ -105,6 +106,35 @@ def __init__(self, logname, logger, metrics, notifier, bgp,
...
+        self._dump_config_event_sock()
+        self._dump_topo_event_sock()
+
+    def _dump_config_event_sock(self):
+        self._notify({'CONFIG_CHANGE': {'success': True,
+            'dps_config': self.meta_dp_state.top_conf}})
+
+    def _dump_topo_event_sock(self):
+        notify_dps = {}
+        root = ""
```
Comment (grafnu) [inline faucet/valves_manager.py]: initialize to None

**Review by grafnu — CHANGES_REQUESTED** [review]

**Comment by anurag6** [comment]
@grafnu pTAL

**Review by grafnu — APPROVED** [review]

**Comment by grafnu** [comment]
You should submit this now, and add the other stuff in another PR.  There is no reason they should be lumped together.  Smaller PR == better.

**Pull request merged**